src/events/aws_sdk.js

```diff
@@ -322,6 +323,64 @@ const dynamoDBEventCreator = {
             });
             break;
 
+        case 'query': {
+            const parametersCopy = { ...parameters };
+            if (config.metadataOnly) {
+                const dataParameters = [
+                    'KeyConditions',
+                    'QueryFilter',
+                    'ExclusiveStartKey',
+                    'ProjectionExpression',
+                    'FilterExpression',
+                    'KeyConditionExpression',
+                    'ExpressionAttributeValues',
+                ];
+                dataParameters.forEach(parameterName => delete parametersCopy[parameterName]);
+            }
+            resource.getMetadataMap().set('Parameters', JSON.stringify(parametersCopy));
+            break;
+        }
+
+        case 'scan': {
+            const parametersCopy = { ...parameters };
+            if (config.metadataOnly) {
+                const dataParameters = [
+                    'ScanFilter',
+                    'ExclusiveStartKey',
+                    'ProjectionExpression',
+                    'FilterExpression',
+                    'ExpressionAttributeValues',
+                ];
+                dataParameters.forEach(parameterName => delete parametersCopy[parameterName]);
```

Comment:
maybe use parametersCopy.filter? also please make sure this doesn't affect original parameters (shouldn't just test locally or something)